### PR TITLE
api: fix error logging

### PR DESCRIPTION
**Pull request by mattrobenolt**
Fixes SENTRY-3WX

https://sentry.io/sentry/sentry/issues/316681526/

### CODE REVIEW

File: src/sentry/tasks/store.py [hunk 1]
```diff
@@ -280,7 +280,7 @@ def save_event(cache_key=None, data=None, start_time=None, event_id=None, **kwar
     except HashDiscarded as exc:
         info_logger.info(
```
Comment (JTCunning) [inline src/sentry/tasks/store.py]: While you're in here, can you change this to `debug`? This is going to harm logging infra as it gets popular.

Comment (mattrobenolt) [inline src/sentry/tasks/store.py]: Do we not ever need to know about these events? This isn't really my call to make since I didn't add this. :) @macqueen

Comment (JTCunning) [inline src/sentry/tasks/store.py]: Discussed in #product. We're good here.

**Review by JTCunning — APPROVED** [review]
🐶

**Pull request merged**